src/plugins/api-token.server.ts

```diff
@@ -0,0 +1,193 @@
+
+export const expiryThreshold = 5 // seconds
+/**
+ * Check whether an access token does not yet exist or if the existing
+ * access token is set to expire soon.
+ * @returns whether the stored access token is about to expire
+ */
+const isNewTokenNeeded = (): boolean => {
+  if (!process.tokenData.accessToken) {
+    log('isNewTokenNeeded: no existing token found')
+    return true
+  }
+
+  const aboutToExpire =
+    process.tokenData.accessTokenExpiry - expiryThreshold <= currTimestamp()
+  log(`isNewTokenNeeded: aboutToExpire=${aboutToExpire}`)
+  return aboutToExpire
+}
+
+/**
+ * Update `tokenData` with  the new access token given the client ID and secret.
+ * @param clientId - the client ID of the application issued by the API
+ * @param clientSecret - the client secret of the application issued by the API
+ */
+const refreshApiAccessToken = async (
+  clientId: string,
+  clientSecret: string
+) => {
+  log('Refreshing access token...')
+  const formData = new URLSearchParams()
+  formData.append('client_id', clientId)
+  formData.append('client_secret', clientSecret)
+  formData.append('grant_type', 'client_credentials')
+
+  const apiService = createApiService()
+  try {
+    const res = await apiService.post<TokenResponse>(
+      'auth_tokens/token',
+      formData
+    )
+    log('Successfully retrieved API token')
+    process.tokenData.accessToken = res.data.access_token
+    process.tokenData.accessTokenExpiry = currTimestamp() + res.data.expires_in
+    log(
+      `Next token expiry for worker expiry=${process.tokenData.accessTokenExpiry}`
+    )
+  } catch (e) {
+    /**
+     * If an error occurs, serve the current request (and any pending)
+     * anonymously and hope it works. By setting the expiry to 0 we queue
+     * up another token fetch attempt for the next request.
+     */
+    error('Unable to retrieve API token, clearing existing token', e)
+    process.tokenData.accessToken = ''
+    process.tokenData.accessTokenExpiry = 0
+    ;(e as AxiosError).message = `Unable to retrieve API token. ${
+      (e as AxiosError).message
+    }`
+    throw e
+  }
+}
+
+process.fetchingMutex = new Mutex()
+
+/**
+ * Get an async function that always returns a valid, automatically-refreshed
+ * API access token.
+ *
+ * The `fetchingMutex` allows all requests on the same process to understand
+ * whether it's necessary for them to request a token refresh or if another
+ * request has already queued the work. If so, they can just await the process-global
+ * promise that will resolve when the api token data refresh request has resolved.
+ *
+ * @param context - the Nuxt context
+ */
+const getApiAccessToken = async (
+  context: Context
+): Promise<string | undefined> => {
+  const { apiClientId, apiClientSecret } = context.$config
+  if (!(apiClientId || apiClientSecret)) {
+    log('No API client secrets found.')
+    return undefined
+  }
+
+  log('We have client information')
```

Comment:
```suggestion
  log('API client secrets found.')
```

For parity with the other log